apps/election-manager/src/components/TallyReportMetadata.tsx

```diff
@@ -0,0 +1,139 @@
+import React from 'react'
+
+import { Dictionary, Election } from '@votingworks/types'
+
+import * as format from '../utils/format'
+import {
+  localeWeedkayAndDate,
+  localeLongDateAndTime,
+} from '../utils/IntlDateTimeFormats'
+import { getLabelForVotingMethod } from '../utils/votingMethod'
+import { VotingMethod } from '../config/types'
+
+import Text from './Text'
+import Table, { TD } from './Table'
+
+interface Props {
+  election: Election
+  generatedAtTime: Date
+  votingWorksBallotCount: number
+  externalBallotCount?: number
+  ballotCountsByVotingMethod?: Dictionary<number>
+}
+
+const TallyReportMetadata: React.FC<Props> = ({
+  election,
+  generatedAtTime,
+  votingWorksBallotCount,
+  externalBallotCount,
+  ballotCountsByVotingMethod,
+}) => {
+  const electionDate = localeWeedkayAndDate.format(new Date(election.date))
+  const generatedAt = localeLongDateAndTime.format(generatedAtTime)
+  const totalBallotCount = votingWorksBallotCount + (externalBallotCount ?? 0)
+  const ballotsByDataSource =
+    externalBallotCount !== undefined ? (
+      <React.Fragment>
+        <h3>Ballots Cast by Data Source</h3>
+        <Table data-testid="data-source-table">
+          <tbody>
+            <tr>
+              <TD as="th">Data Source</TD>
+              <TD as="th" textAlign="right">
+                Number of Ballots Cast
+              </TD>
+            </tr>
+            <tr>
+              <TD>VotingWorks Data</TD>
+              <TD textAlign="right">{format.count(votingWorksBallotCount)}</TD>
```

Comment:
same comment as previous PR, would suggest not using `votingworks` as variable names, more something like `internal`.